Document/0x05b-Basic-Security_Testing.md

```diff
@@ -66,6 +66,62 @@ Once you've configured the network and established a connection between the test
+Network security configuration uses an XML file where the app specify which CAs certificates will be trusting.
+
+res/xml/network_security_config.xml:
+```shell
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <base-config>
+        <trust-anchors>
+            <certificates src="@raw/extracas"/>
+            <certificates src="system"/>
+        </trust-anchors>
+    </base-config>
+</network-security-config>
+```
+The apps must include an entry in the manifest file to point to the network security configuration file.
+```shell
+<?xml version="1.0" encoding="utf-8"?>
+<manifest ... >
+    <application android:networkSecurityConfig="@xml/network_security_config"
+                    ... >
+        ...
+    </application>decomp
+</manifest>
+```
+
+In order to intercept an application's traffic which Android version is 7 and above, you must follow the steps below:
+
+- Decompile the app using decompiling tools.[Manual static Analysis] provides details about decompiling (https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05b-Basic-Security_Testing.md#manual-static-analysis)
+- Make the application trust the proxy's certificate by creating a network security configuration with the giving certificate as explained above
+- Repackage the app.[Android developer documentation](https://developer.android.com/studio/publish/app-signing#signing-manually) explains how it's done. 
+
+Note that even if this method is quite simple its major drawback is that you have to apply this operation for each application you want to evaluate which is redundant and annoying.
+
+##### Adding the Proxy's certificate among system trusted CAs
+
+In order to avoid the obligation of configuring the Network Security Configuration for each application, we must force the Device to accept the proxy's certificate as one of the system strusted certificates, to do so follow the steps below:
+
+- Making the system files writable which requires rooting the Device.Find instructions on how to [root](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05b-Basic-Security_Testing.md#connecting-to-an-android-virtual-device-avd-as-root) your device later in this chapter.Run 'mount' command to make sure the /system is writable, if it still not the case run the following command 'mount -o rw,remount -t ext4 /system'
+- Preparing the proxy's CA cerificates to match system certificates format.Export the proxy's certificates in der format (this is based on burp suites case) then run the following commands:
+```shell
+openssl x509 -inform DER -in cacert.der -out cacert.pem  
+openssl x509 -inform PEM -subject_hash_old -in cacert.pem |head -1  
+mv cacert.pem <hash>.0
+```
+- Finally, copy the <hash>.0 file in /system/etc/security/cacerts then run this command 
+```shell 
+chmod 644 <hash>.0
+```
+
+By following the steps presented above you make sure that any application will trust the proxy's certificates, so you can intercept its traffic.
```

Comment:
By following the steps described above, you allow any application to trust the proxy's certificates, which allows  you to intercept its traffic.

-> I was restructuring the sentence a bit. Is the meaning still correct?

Reply:
It makes sense to me.